app/code/Magento/Backup/Model/ResourceModel/View/GetViewsBackup.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Backup\Model\ResourceModel\View;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Backup\Db\BackupInterface;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+
+/**
+ * Class GetViewsBackup
+ */
+class GetViewsBackup
+{
+    /**
+     * @var GetListViews
+     */
+    private $getListViews;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $connection;
+
+    /**
+     * @param GetListViews $getListViews
+     * @param ResourceConnection $resourceConnection
+     */
+    public function __construct(
+        GetListViews $getListViews,
+        ResourceConnection $resourceConnection
+    ) {
+        $this->getListViews = $getListViews;
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * @param BackupInterface $backup
+     */
+    public function execute(BackupInterface $backup)
+    {
+        $views = $this->getListViews->execute();
+
+        foreach ($views as $view) {
+            $backup->write($this->getViewHeader($view));
+            $backup->write($this->getDropViewSql($view));
+            $backup->write($this->getShowCreateView($view));
+        }
+    }
+
+    /**
+     * @return AdapterInterface
+     */
+    private function getConnection()
+    {
+        if (!$this->connection) {
+            $this->connection = $this->resourceConnection->getConnection('backup');
+        }
+
+        return $this->connection;
+    }
+
+    /**
+     * @param string $viewName
+     * @return string
+     */
+    private function getShowCreateView($viewName)
+    {
+        $quotedViewName = $this->getConnection()->quoteIdentifier($viewName);
+        $query = 'SHOW CREATE VIEW ' . $quotedViewName;
```

Comment:
creating View we have to use this syntax
```
                SQL SECURITY INVOKER
```

https://dev.mysql.com/doc/refman/5.6/en/stored-programs-security.html
```
For a stored routine or view, use SQL SECURITY INVOKER in the object definition when possible so that it can be used only by users with permissions appropriate for the operations performed by the object.
```

In opposite case, MySQL user working with Magento would not be possible to invoke the View after creation.
This is especially affect our Cloud deployment

Reply:
@maghamed 
If this VIEW was created with SQL SECURITY INVOKER declaration - SHOW CREATE VIEW will show sql statement with this declaration also.
![image](https://user-images.githubusercontent.com/15912461/53478078-b2b4e280-3a7e-11e9-8a22-deb37e70df6e.png)

If no - then with SQL SECURITY DEFINER
![image](https://user-images.githubusercontent.com/15912461/53478143-d6782880-3a7e-11e9-9811-7d165271dda6.png)
 
So it's ok with current implementation

Reply:
Agree, we already use `SQL SECURITY INVOKER` at the time of View creation on MSI, so that backup with just pick this setting and at the time of DB restore we keep that setting